src/EventListener/ProductPublisher.php

```diff
@@ -49,10 +88,50 @@ public function onFlush(OnFlushEventArgs $event)
      */
     public function postFlush(PostFlushEventArgs $event)
     {
-        foreach ($this->scheduledProducts as $product) {
+        foreach ($this->scheduledInserts as $product) {
             $this->eventBus->handle(ProductCreated::occur($product));
         }
 
-        $this->scheduledProducts = [];
+        $this->scheduledInserts = [];
+
+        foreach ($this->scheduledUpdates as $product) {
+            $this->eventBus->handle(ProductUpdated::occur($product));
+        }
+
+        $this->scheduledUpdates = [];
+
+        foreach ($this->scheduledDeletions as $product) {
+            $this->eventBus->handle(ProductDeleted::occur($product));
+        }
+
+        $this->scheduledDeletions = [];
+    }
+
+    /**
+     * @param $entity
+     *
+     * @return ProductInterface|\Sylius\Component\Product\Model\ProductInterface|\Sylius\Component\Resource\Model\TranslatableInterface
+     */
+    protected function getProductFromEntity($entity): ?ProductInterface
+    {
+        if ($entity instanceof ProductVariantInterface) {
+            $entity = $entity->getProduct();
```

Comment:
We can just return the value here.

Reply:
Well, I though that having a single return is better.
Also, I think switching this to a `switch` statement